react/features/recording/components/Recording/native/StartRecordingDialog.js

```diff
@@ -85,6 +67,31 @@ class StartRecordingDialog extends AbstractStartRecordingDialog<Props> {
         this._onSubmit() && goBack();
     }
 
+    isStartRecordingDisabled: () => boolean;
+
+    /**
+     * Disables start recording button.
+     *
+     * @returns {boolean}
+     */
+    isStartRecordingDisabled() {
+        const { _fileRecordingsServiceEnabled, _isDropboxEnabled } = this.props;
+        const { isTokenValid, isValidating } = this.state;
+
+        // Start button is disabled if recording service is only shown;
+        // When validating dropbox token, if that is not enabled, we either always
+        // show the start button or, if just dropbox is enabled, start button
+        // is available when there is token.
+        if (_fileRecordingsServiceEnabled) {
+            return isValidating;
+        } else if (_isDropboxEnabled) {
+            return isTokenValid;
```

Comment:
Did you figure out why this is different than the web case where we return `! isTokenValid `?

Reply:
Yep, the bug was we were not updating the header after mounting. I still recommended we make the logic more readable, so that's what we did :-)